src/server/pegasus_write_service.cpp

```diff
@@ -36,111 +124,36 @@ pegasus_write_service::pegasus_write_service(pegasus_server_impl *server)
       _server(server),
       _impl(new impl(server)),
       _batch_start_time(0),
-      _cu_calculator(server->_cu_calculator.get())
+      _cu_calculator(server->_cu_calculator.get()),
+      _put_counter(METRIC_put_requests.instantiate(replica_metric_entity())),
```

Comment:
Will the counter_ptr/percentile_ptr get any error if not initialized in c'tor? It will be helpful to prevent missing initialized.

Reply:
Normally instantiations for all metric types will not lead to error. Once instantiation fails, the process will crash which means there must have been some bug. For example, the metric entity is uninitialized.